### PR TITLE
New package: openturns

### PR DESCRIPTION
**Pull request by bfovet**
This PR introduces the OpenTURNS UQ library, along with its dependencies and new versions for NLopt and Ceres Solver.
There might still be some work left to do with variants and dependencies.

### CODE REVIEW

File: var/spack/repos/builtin/packages/openturns/package.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    depends_on('libxml2', when='+xml')
+    depends_on('intel-tbb', when='+tbb')
+    depends_on('py-matplotlib', when='+matplotlib')
+    depends_on('r', when='+stats')
+    depends_on('py-psutil', when='+coupling')
+    depends_on('doxygen', when='+doc')
+    depends_on('py-sphinx', when='+doc')
+    depends_on('py-numpydoc', when='+doc')
+    depends_on('py-nbconvert', when='+doc')
+    depends_on('py-ipython', when='+doc')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/openturns/package.py]: Most Python dependencies should be type=build/run. Doc dependencies should be type=build only

File: var/spack/repos/builtin/packages/optpp/package.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    def configure_args(self):
+        spec = self.spec
+
+        args = [
+            '--enable-shared'
+        ]
+
+        if '+mpi' in spec:
+            args.append('--enable-mpi')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/optpp/package.py]: I would explicitly disable this in an else clause

File: var/spack/repos/builtin/packages/optpp/package.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+        if '+blas' in spec:
+            args.append('--with-blas=%s' % spec['blas'].libs)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/optpp/package.py]: Doesn't this return a list?

**Comment by bfovet** [comment]
> This PR would be easier to review if it were split into 8 separate PRs

You're right, let's take it slow. I'll open PRs for each package, and take your remarks (thanks btw!) into account.

**Pull request closed without merging**